examples/bert/src/bert_layers.py

```diff
@@ -474,7 +481,18 @@ def forward(
         #        hidden_states[ntokens,hidden] -> hidden_states[ntokens_unpad,hidden]
         hidden_states, indices, cu_seqlens, _ = unpad_input(
             hidden_states, attention_mask_bool)
+
         # Add alibi matrix to extended_attention_mask
+        if self._current_alibi_size < seqlen or self.alibi is None:
+            # Build/rebuild the alibi tensor if needed
+            self.alibi = self.build_alibi_tensor(
+                n_heads=self.num_attention_heads,
+                max_token_length=seqlen,
+                device=hidden_states.device)
+            self._current_alibi_size = int(seqlen)
+        if self.alibi.device != hidden_states.device:
+            # Device catch-up
+            self.alibi = self.alibi.to(hidden_states.device)
         alibi_bias = self.alibi[:, :, :seqlen, :seqlen]  # type: ignore
         alibi_attn_mask = extended_attention_mask[:, :, :, :seqlen] + alibi_bias
 
```

Comment:
This is great! I like that the ALiBi tensor is built to the right size on the fly. This seems like the correct way to do it

Reply:
It also means that the mcloud configs don't have to deal with anything other than the maximum sequence length =)